wallets/core/src/namespaces/evm/actions.ts

```diff
@@ -0,0 +1,97 @@
+import type { EIP1193EventMap } from './eip1193.js';
+import type { EvmActions, ProviderApi } from './types.js';
+import type { Context, Subscriber } from '../../hub/namespaces/mod.js';
+import type { SubscriberCleanUp } from '../../hub/namespaces/types.js';
+import type { CaipAccount } from '../../types/accounts.js';
+import type { FunctionWithContext } from '../../types/actions.js';
+
+import { AccountId } from 'caip';
+
+import { recommended as commonRecommended } from '../common/actions.js';
+
+import { CAIP_NAMESPACE } from './constants.js';
+import { getAccounts, switchOrAddNetwork } from './utils.js';
+
+export const recommended = [...commonRecommended];
+
+export function connect(
+  instance: () => ProviderApi
+): FunctionWithContext<EvmActions['connect'], Context> {
+  return async (_context, chain) => {
+    const evmInstance = instance();
+
+    if (!evmInstance) {
+      throw new Error(
+        'Do your wallet injected correctly and is evm compatible?'
+      );
+    }
+
+    if (chain) {
+      await switchOrAddNetwork(evmInstance, chain);
+    }
+
+    const chainId = await evmInstance.request({ method: 'eth_chainId' });
+
+    const result = await getAccounts(evmInstance);
+
+    const formatAccounts = result.accounts.map(
+      (account) =>
+        AccountId.format({
+          address: account,
+          chainId: {
+            namespace: CAIP_NAMESPACE,
+            reference: chainId,
+          },
+        }) as CaipAccount
+    );
+
+    return {
+      accounts: formatAccounts,
+      network: result.chainId,
+    };
+  };
+}
+
+export function changeAccountSubscriber(
```

Comment:
I couldn't find related method for changeNetworkSubscriber. How do you handle chain subscriber?

Reply:
Good question. Phantom, our first provider, doesn't have this functionality. so the plan is keeping the scope of this PR limited to what we need for Hub and Phantom. We will add missing functionality to core, when we are migrating our providers to Hub. I mean, this PR is going to have infrastructure and a base, to be able add more functionality on top of it.

Reply:
As we talked offline, it seems phantom has support for `chainChanged`. I will look into that in my other PR.